orquesta/tests/fixtures/workflows/native/cycle-fork.yaml

```diff
@@ -0,0 +1,46 @@
+---
+version: '1.0'
+
+description: A sample workflow with a fork that stems from a cycle.
+
+vars:
+  - cheer: false
+  - work: false
+
+tasks:
+  # This init task is required to tell the workflow engine where to start.
+  # Otherwise with just a cycle, the workflow engine cannot tell where to start.
+  init:
+    action: core.noop
+    next:
+      - do: query
+
+  # This is the start of the cycle in the workflow.
+  query:
+    action: core.noop
+    next:
+      - when: <% succeeded() %>
+        publish: cheer=<% result() %> work=<% result() %>
+        do: decide_cheer, decide_work
+
+
+  # The branch under the decide_work task will cycle back to the query task.
+  decide_work:
+    next:
+      - when: <% ctx().work %>
+        do: notify_work, toil
+  notify_work:
+    action: core.noop
+  toil:
+    action: core.echo message="This is hard work."
+    next:
+      - do: query
```

Comment:
The workflow here is infinity loop, right?

Reply:
@jinpingh If you look at the workflow definition and the unit test closely, it is not an infinitely loop practically. In the unit test, I'm controlling the result that is returned by the query task which controls when the loop terminates.

Reply:
Understood that unit test takes care of when the loop terminates. Just want to make sure that I understand how this fork working.  :)

Reply:
Gotcha.  You are asking the right questions.